app/main.py

```diff
@@ -1,6 +1,23 @@
 from typing import Callable
 
-
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    storage = {}
```

Comment:
The `storage` dictionary is used to cache results of function calls. However, using mutable types like lists or dictionaries as keys in a dictionary will raise a `TypeError`. Ensure that the arguments to the cached function are immutable (e.g., tuples) to avoid this issue.